Docs - IOTA - Improve Getting Started With Rust (#1868)
iotaledger/wallet.rs: documentation/docs/getting_started/rust.mdx

- system environment variables
- getting started
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Getting Started with Rust

> The wallet.rs library is published on [crates.io](https://crates.io/crates/iota-wallet/versions). You can also download the source code from the [official GitHub repository](https://github.com/iotaledger/wallet.rs).

Reviewer: We only started publishing the stardust versions, because in production we still have not published dependencies which make it impossible for us to publish to crates.io

Author: ```suggestion
You download the source code for the wallet.rs library from the [official GitHub repository](https://github.com/iotaledger/wallet.rs).
```
Sorry about that, I got confused with the 0.1 version that's published. I think this ought to fix it.  Thanks for the clarification :)

[_no_std_](https://docs.rust-embedded.org/book/intro/no-std.html) is not currently supported, but we are working on it, and we will provide it as a feature once the new implementation is ready.

### Dependencies

[_cmake_](https://cmake.org/documentation/) and [_openssl_](https://www.openssl.org/docs/) are required to run Rust. To run the build process successfully using Cargo, you may need install additional build tools onto your system.

<Tabs>
<TabItem value={'windows'} label={'Windows'}>

You can download _cmake_ from the [official cmake website](https://cmake.org/download/).
You can install _openssl_ with [vcpkg](https://github.com/microsoft/vcpkg) or
[chocolatey](https://chocolatey.org/).

<Tabs>
<TabItem value={'vcpkg'} label={'vcpkg'}>

```bash
./vcpkg.exe install openssl:x64-windows
./vcpkg.exe integrate install
# you may want to add this to the system environment variables since you'll need it to compile the crate
set VCPKGRS_DYNAMIC=1
```

</TabItem>
<TabItem value={'chocolatey'} label={'chocolatey'}>

```bash
choco install openssl
# you may need to set the OPENSSL_ROOT_DIR environment variable
set OPENSSL_ROOT_DIR="C:\Program Files\OpenSSL-Win64"
```

> </TabItem>

Reviewer 2: Unrelated to this PR but can we make chocolatey the default and first entry? I think the chocolatey user base is substantially bigger than the vcpkg one 😅

</Tabs>
</TabItem>
<TabItem value={'macOS'} label={'macOS'}>

You can install _cmake_ and _openssl_ with [_Homebrew_](https://docs.brew.sh/) by running the following
commands:

```bash
brew install cmake
brew install openssl@1.1
# you may want to add this to your .zshrc or .bashrc since you'll need it to compile the crate
OPENSSL_ROOT_DIR=$(brew --prefix openssl@1.1)
```

</TabItem>
<TabItem value={'linux'} label={'Linux'}>

You can install _cmake_ and _openssl_ with your distro's package manager or download them from their websites.
On Debian and Ubuntu, you will also need the
[_build-essential_](https://packages.debian.org/sid/build-essential) package.

</TabItem>
</Tabs>

## Use the Library

### Add wallet.rs as a Dependency

If you simply want to use the library, you only need to add this lines your _Cargo.toml_ file:

```
[dependencies]
iota-wallet = { git = "https://github.com/iotaledger/wallet.rs", branch = "production" }
```

### Enable Asynchronous Functionality

The example below requires asynchronous functionality. You can enable it by adding the following line as a dependency:

```
tokio = { version = "1", features = ["full"] }
```

### Initialize the Library

> To use the library, you first need to create an [`AccountManager`](asd), as shown on lines 7 through 10 in the following example.

Reviewer: `asd`?

Author: placeholder, sorry about that. Fixed!
